jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java

```diff
@@ -129,21 +125,41 @@ private static final Object resolveConstantDynamic(long j9class, String name, St
 		 */
 		if (bsmTypeArgCount < 1 || MethodHandles.Lookup.class != bsm.type().parameterType(0)) {
 			/*[MSG "K0A01", "Constant_Dynamic references bootstrap method '{0}' does not have java.lang.invoke.MethodHandles.Lookup as first parameter."]*/
+/*[IF OPENJDK_METHODHANDLES]*/
+			throw new BootstrapMethodError(Msg.getString("K0A01", bsm)); //$NON-NLS-1$
+/*[ELSE] OPENJDK_METHODHANDLES*/
 			throw new BootstrapMethodError(Msg.getString("K0A01", bsm.getMethodName())); //$NON-NLS-1$
```

Comment:
What is the difference in output between the `bsm`'s toString() and `bsm.getMethodName()`?

Reply:
`bsm` is a `MethodHandle`. OpenJDK MH doesn't have `getMethodName`. So, I replaced it with just `bsm` for the time being. [MethodHandleInfo.getName()](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/invoke/MethodHandleInfo.html#getName()) is an alternative, which would return the method name.
```suggestion
			throw new BootstrapMethodError(Msg.getString("K0A01", Lookup.IMPL_LOOKUP.revealDirect(bsm).getName())); //$NON-NLS-1$
/*[ELSE] OPENJDK_METHODHANDLES*/
			throw new BootstrapMethodError(Msg.getString("K0A01", bsm.getMethodName())); //$NON-NLS-1$
```

Reply:
Changed to `Lookup.IMPL_LOOKUP.revealDirect(bsm).getName()` (OpenJDK), which is equivalent to `MH.getMethodName()` (OpenJ9).